tests/test_vault.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"vault secrets tests"
+
+import base64
+import os
+import shutil
+import tempfile
+import unittest
+from time import sleep
+
+import docker
+import hvac
+
+from kapitan.refs.base import RefController, RefParams, Revealer
+from kapitan.refs.secrets.vaultkv import VaultSecret, vault_obj
+
+# Create temporary folder
+REFS_HOME = tempfile.mkdtemp()
+REF_CONTROLLER = RefController(REFS_HOME,target_name='minikube-mysql',
+                               inventory_path=os.path.join('.','examples','kubernetes','inventory'))
+REVEALER = Revealer(REF_CONTROLLER)
+# Create Vault docker container
+client = docker.from_env()
+env = {'VAULT_LOCAL_CONFIG':'{"backend": {"file": {"path": "/vault/file"}}, "listener":{"tcp":{"address":"0.0.0.0:8200","tls_disable":"true"}}}'}
+vault_container= client.containers.run(image='vault',cap_add=['IPC_LOCK'],ports={'8200':'8200'},
+                                       environment=env,detach=True,remove=True,command='server')
+sleep(4)
```

Comment:
Sleep here is quite unreliable to wait until the container is up. On different CI/CDs it might take longer than that and then the test will fail.
Use `docker inspect` in a loop to see if the container is actually in Ready state, and if not, sleep 1s again each time.
The docker bindings also give you the "status" https://docker-py.readthedocs.io/en/stable/containers.html so you can use that too to see if the container is Ready. :)

Reply:
This makes more sense, thank you, I actually changed it from 1s -> 4s (just in case) but checking the status is generic method :)